app/assets/javascripts/actions/projectBoard.js

```diff
@@ -53,3 +68,25 @@ export const fetchProjectBoard = projectId =>
       return dispatch(errorRequestProjectBoard(error));
     }
   };
+
+export const closeSearch = () =>
+  async (dispatch, getState, {}) => {
+    try {
+      dispatch(closeSearchSuccess());
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+export const search = (keyWord, projectId) =>
+  async (dispatch, getState, { ProjectBoard }) => {
+    try {
+      const search = await ProjectBoard.searchStories(keyWord, projectId);
+
+      dispatch(searchStoriesSuccess(search.keyWord));
+      dispatch(receiveStories(search.result, 'search'));
+    }
+    catch (error) {
+      console.error(error)
```

Comment:
We probably should do more than logging in case search fails, but the structure of this action brings me to a question:

If the user types something in the search and hits enter, will there be some visual feedback while waiting? Or will nothing be shown until the search finishes and, if it fails, nothing will be shown at all? We should improve that, like showing the search result column with a "Loading results..." or something like that and, if the search fail, we add a message asking the user to try again.